htlcswitch/hop/payload.go

```diff
@@ -73,3 +117,47 @@ func NewPayloadFromReader(r io.Reader) (*Payload, error) {
 func (h *Payload) ForwardingInfo() ForwardingInfo {
 	return h.FwdInfo
 }
+
+// ValidateParsedPayloadTypes checks the types parsed from a hop payload to
+// ensure that the proper fields are either included or omitted. The finalHop
+// boolean should be true if the payload was parsed for an exit hop. The
+// requirements for this method are described in BOLT 04.
+func ValidateParsedPayloadTypes(parsedTypes tlv.TypeSet, finalHop bool) error {
+	_, hasAmt := parsedTypes[record.AmtOnionType]
+	_, hasLockTime := parsedTypes[record.LockTimeOnionType]
+	_, hasNextHop := parsedTypes[record.NextHopOnionType]
+
+	switch {
+
+	// All hops must include an amount to forward.
+	case !hasAmt:
+		return ErrInvalidPayload{
+			Type:     record.AmtOnionType,
+			Name:     "amount to forward",
```

Comment:
Not sure about the string here. Would an enum be overkill? The string here and in the unit test must be kept equal  manually and maybe more matching on the string will happen.

Reply:
done, removed string and only print type in log